ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -342,6 +356,240 @@ objects:
+                  $@
+          }
+
+          # Machine API won't let the user to destroy the node which runs the controller
+          echo "Finding two masters to destroy"
+          MAPI_POD=$(oc get pod -l k8s-app=controller -n openshift-machine-api --no-headers -o name)
+          SURVIVING_MASTER_NODE=$(oc get ${MAPI_POD} -n openshift-machine-api -o jsonpath='{.spec.nodeName}')
+          mapfile -t MASTER_NODES_TO_REMOVE < <(oc get nodes -l node-role.kubernetes.io/master= -o name | grep -v "${SURVIVING_MASTER_NODE}")
+          MASTER_MACHINES_TO_REMOVE=()
+          for master in ${MASTER_NODES_TO_REMOVE[@]}
+          do
+            MASTER_MACHINES_TO_REMOVE+=($(oc get ${master} -o jsonpath='{.metadata.annotations.machine\.openshift\.io\/machine}' | cut -d '/' -f 2))
+          done
+
+          echo "Destroy two masters"
+          # Scale down etcd quorum guard
+          oc scale --replicas=0 deployment.apps/etcd-quorum-guard -n openshift-machine-config-operator
+
+          for machine in ${MASTER_MACHINES_TO_REMOVE[@]}
+          do
+            retry 10 oc --request-timeout=5s -n openshift-machine-api delete machine ${machine}
+          done
+
+          echo "Confirm meltdown"
+          sleep 30
+          oc --request-timeout=5s get nodes && exit 1
+
+          echo "Upload a script to restore single node cluster"
+          cat > /tmp/install_recovery_scripts.sh <<'SCRIPT_EOF'
+        #!/bin/bash
+        set -x
+        if [[ $EUID -ne 0 ]]; then
+          echo "This script must be run as root"
+          exit 1
+        fi
+
+        PREFIX="https://raw.githubusercontent.com/vrutkovs/openshift-recovery/unify-fixes"
+
+        curl -s $PREFIX/bin/etcd-member-recover.sh -o /usr/local/bin/etcd-member-recover.sh
+        chmod 755 /usr/local/bin/etcd-member-recover.sh
+
+        curl -s $PREFIX/bin/etcd-snapshot-restore.sh -o /usr/local/bin/etcd-snapshot-restore.sh
+        chmod 755 /usr/local/bin/etcd-snapshot-restore.sh
+
+        curl -s $PREFIX/bin/openshift-recovery-tools -o /usr/local/bin/recovery-tools
+
+        curl -s $PREFIX/bin/tokenize-signer.sh -o /usr/local/bin/tokenize-signer.sh
+        chmod 755 /usr/local/bin/tokenize-signer.sh
+        SCRIPT_EOF
+          chmod +x /tmp/install_recovery_scripts.sh
+
+          retry 10 \
+            scp -o StrictHostKeyChecking=no -o ProxyCommand="ssh -A -o StrictHostKeyChecking=no \
+                -o ServerAliveInterval=30 -W %h:%p core@${BASTION_HOST}" \
+                /tmp/install_recovery_scripts.sh core@"${SURVIVING_MASTER_NODE}":/tmp
+          echo "Run recovery scripts installer"
+          bastion_ssh core@${SURVIVING_MASTER_NODE} "sudo -i /tmp/install_recovery_scripts.sh"
+
+          echo "Restore single etcd"
+          bastion_ssh core@${SURVIVING_MASTER_NODE} "sudo -i /usr/local/bin/etcd-snapshot-restore.sh"
+
+          echo "Wait for API server to come up"
+          retry 30 oc get nodes
```

Comment:
Not part of the test framework that I can review, but if your etcd cluster is down to one node (because you killed two of three control-plane nodes), then how is the remaining etcd still functioning?  I'd have expected it to be freaking out about having lost quorum and refusing to take possibly-split-brained actions.  If there's some quick explanation for how this works, I'm very curious.  If the explanation is longer, we should probably skip it to avoid distracting from the test script itself.

Reply:
>if your etcd cluster is down to one node (because you killed two of three control-plane nodes), then how is the remaining etcd still functioning?

`etcd-snapshot-restore.sh` without params would restore this etcd forming a one node cluster - https://github.com/openshift/machine-config-operator/blob/master/templates/master/00-master/_base/files/usr-local-bin-openshift-recovery-tools-sh.yaml#L141-L168.

Added a bit better comment for this in b99911295